### PR TITLE
fix(dropdown): guarantee initial selected item is tracked locally, collect selectItem from render calls, manually force reselection even where selected item exists to notify Dropdown consumer of current selection state

### PR DESCRIPTION
**Pull request by Pallandor**
fix(dropdown): guarantee initial selected item is tracked locally, collect selectItem from render calls, manually force reselection even where selected item exists to notify Dropdown consumer of current selection state

---
This change is [Reviewable](https://reviewable.io/reviews/qantasairways/runway/77)

### CODE REVIEW

File: src/components/Dropdown/index.js [hunk 1]
```diff
@@ -309,6 +319,7 @@ export default class Dropdown extends React.Component {
             setFocus,
             downshiftProps
           };
+          this.collectSelectItem(renderProps.downshiftProps.selectItem);
```
Comment (rachmulvey) [inline src/components/Dropdown/index.js]: Would it make more sense to call this in `componentDidUpdate`?

Comment (Pallandor) [reply src/components/Dropdown/index.js]: Ah I experimented with that initially. But on some lifecycle calls, downshiftProps was strangely not populated so I couldn't collect selectItem. This way I'm guaranteed a fresh reference always😄

File: src/components/Dropdown/__tests__/Dropdown.spec.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        );
+        const nextItems = [pear, orange];
+        component.setProps({ items: nextItems });
+        component.update();
+        const input = component.find('input');
+        expect(input.props().value).toEqual(orange.name);
+      });
+    });
+    describe('when current selected is does NOT exist in items AND defaultItemWhenNoneSelected is valid AND still exists in items', () => {
+      it.only('should keep select defaultItemWhenNoneSelected', () => {
```
Comment (Minaparnian) [inline src/components/Dropdown/__tests__/Dropdown.spec.js]: Please remove only.

Comment (Pallandor) [reply src/components/Dropdown/__tests__/Dropdown.spec.js]: Agreed and done!

**Pull request merged**